crypto/lhash/lhash.c

```diff
@@ -72,11 +72,17 @@ void *OPENSSL_LH_insert(OPENSSL_LHASH *lh, void *data)
     unsigned long hash;
     OPENSSL_LH_NODE *nn, **rn;
     void *ret;
-
+    int val = 0;
+    
     lh->error = 0;
     if (lh->up_load <= (lh->num_items * LH_LOAD_MULT / lh->num_nodes))
-        expand(lh);
-
+        val = expand(lh);
+    
+    if (val != 0) {
```

Comment:
In function 'expand', return 0 indicate success and return  -1/1 indicate failure.
In your note, do I need to add some success process in parentheses cause you use '!expand(lh)' expression? Otherwise, maybe you want me to change the return semanteme in expand.

if ((lh->up_load <= (lh->num_items * LH_LOAD_MULT / lh->num_nodes)) && !expand(lh)) {
    /*success process ?*/
}